src/atoms/Radio/Radio.web.js

```diff
@@ -1,5 +1,50 @@
-import React from 'react';
+import React, { useState, useCallback, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { getVariantColorKeys } from '../../_helpers/theme';
+import RadioButtonContext from './RadioContext';
+import RadioOption from './RadioOption';
+import Fieldset from './Fieldset';
 
-const RadioButton = () => <div>WIP</div>;
+const Radio = ({ value, onChange, defaultValue, variantColor, size, children }) => {
+  const [selected, setSelected] = useState(value || defaultValue);
 
-export default RadioButton;
+  const onValueChange = useCallback(
+    (newValue) => {
+      if (!value) {
+        setSelected(newValue);
+      }
+      onChange(newValue);
+    },
+    [onChange, value],
+  );
+
+  const contextValue = useMemo(
+    () => ({ value: selected, onChange: onValueChange, variantColor, size }),
+    [onValueChange, selected, variantColor, size],
+  );
+
+  return (
+    <RadioButtonContext.Provider value={contextValue}>
```

Comment:
There's some disparity between calling components with prefix "Radio" and "RadioButton". I think we should stick with "Radio" everywhere which would make this `RadioContext`.

Reply:
Check other places like `useRadioContext` instead of `useRadioButtonContext`.